server/neptune/workflows/internal/deploy/revision/revision.go

```diff
@@ -31,18 +31,24 @@ type Queue interface {
 	Push(terraform.DeploymentInfo)
 	GetLockState() queue.LockState
 	SetLockForMergedItems(ctx workflow.Context, state queue.LockState)
+	Scan() []terraform.DeploymentInfo
+}
+
+type Worker interface {
```

Comment:
nit: This doesn't need to know about Worker, you can probably rename this to something more relevant to the method like `DeploymentStore`